src/test/scala/com/mwz/sonar/scala/ScalaTest.scala

```diff
@@ -6,17 +6,49 @@ import org.scalatest.{FlatSpec, Matchers}
 import org.sonar.api.config.internal.MapSettings
 
 class ScalaTest extends FlatSpec with Matchers {
-  "getScalaVersion" should "return the available version" in {
+  "getScalaVersion" should "return the available version, if properly set" in {
     val conf = new MapSettings()
-      .setProperty("sonar.scala.version", "123")
+      .setProperty("sonar.scala.version", "2.11.11")
       .asConfig()
 
-    Scala.getScalaVersion(conf) shouldBe "123"
+    Scala.getScalaVersion(conf).toString shouldBe "2.11.11"
   }
 
-  it should "return the default value if not set" in {
+  it should "return the default version, if the property is not set" in {
     val conf = new MapSettings().asConfig()
-    Scala.getScalaVersion(conf) shouldBe "2.11.0"
+    Scala.getScalaVersion(conf).toString shouldBe "2.11.0"
+  }
+
+  it should "be able to parse a milestone version" in {
+    val conf = new MapSettings()
+      .setProperty("sonar.scala.version", "2.13.0-M3")
+      .asConfig()
+
+    Scala.getScalaVersion(conf).toString shouldBe "2.13.0-M3"
+  }
+
+  it should "be able to parse a version with an empty patch" in {
+    val conf = new MapSettings()
+      .setProperty("sonar.scala.version", "2.12.")
+      .asConfig()
+
+    Scala.getScalaVersion(conf).toString shouldBe "2.12."
+  }
+
+  it should "be able to parse a version without patch" in {
```

Comment:
So this is failing, but we want to make it pass.
The only idea I have is to use our own regex (to be precisely this one `"""(\d+)\.(\d+)(?:\.(.+))?""".r`). Do you like it? or what other ideas do you have?

```scala
scala> val r = """(\d+)\.(\d+)(?:\.(.+))?""".r
r: scala.util.matching.Regex = (\d+)\.(\d+)(?:\.(.+))?

scala> r.unapplySeq("2.12.6")
res6: Option[List[String]] = Some(List(2, 12, 6))

scala> r.unapplySeq("2.12")
res7: Option[List[String]] = Some(List(2, 12, null))
```

Reply:
Yeah, I think your regex is ok 👍 

You can also probably reuse the `parse` function from scaliform with a small change, e.g.:

```scala
import scalariform.ScalaVersion
import scalariform.utils.Utils._

val VersionPattern = """(\d+)\.(\d+)(?:\.(.+))?""".r

def parse(s: String): Option[ScalaVersion] = s match {
  case VersionPattern(major, minor, _) =>
    for {
      major <- major.toIntOpt
      minor <- minor.toIntOpt
    } yield ScalaVersion(major, minor)
  case _ ⇒
    None
}
```

But obviously whenever you want to use the string version, you will need to concatenate the major and monir versions yourself since the `ScalaVersion.toString` returns an extra `.` at the end.